### PR TITLE
feat: Spelling change of `Subtype.val` in `LinearIndependent`

### PR DESCRIPTION
**Pull request by apnelson1**
To state that the vectors in `s : Set V` are linearly independent, one must write something equivalent to 
`LinearIndependent R (Subtype.val : s → V)`. This is currently done inconsistently, with spellings such as 
`LinearIndependent R ((↑) : s → V)` and `LinearIndependent R (fun x : s => x)` appearing in different places. 
All these have the problem that `LinearIndependent R Subtype.val` is what shows up in the docs and infoview, 
so it is hard to see what set is being talked about, or to rewrite `s` to some equal set. 

This PR defines a new abbreviation `s.incl := Subtype.val  : s → V`, and changes all such expressions to `LinearIndependent R s.incl`. This spelling means that the and pretty-printer will show `s` explicitly, and rewriting is now possible. 

We do the same for `AffineIndependent`, and also change `LinearIndependent R (f ∘ ((↑)  : s → V))` to `LinearIndependent R (s.restrict f)` in a few places, which has the same advantages. We put a comment in the `LinearIndependent.lean` docstring explaining this. 

There are presumably many other places where changing `Subtype.val` to `s.incl` for some `s` could be an improvement in the pp/docs. This PR sticks to Affine/Linear independence. 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by eric-wieser** [comment]
This seems to have picked up some unrelated Matroid changes

File: Mathlib/Data/Matroid/Basic.lean
Comment (eric-wieser) [inline Mathlib/Data/Matroid/Basic.lean:1]: Presumably these change also don't belong in this PR.

**Comment by apnelson1** [comment]
> This seems to have picked up some unrelated Matroid changes

Thanks - I think this is now fixed.

**Comment by eric-wieser** [comment]
> A suggested future change (see [this Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Spelling.20of.20Subtype.2Eval.20in.20LinearIndependent)) is to define a new function `s.incl` as `s.restrict id`, to further standardize this convention. This would be fairly easy to enact with a global search/replace.

I think we should instead define
```lean
/-- An alias for `Subtype.val` from sets, which takes `s` as an explicit argument to make it clear what the set in question is. -/
abbrev Set.incl (s : Set α) : s → α := Subtype.val
```
so that no existing proofs have to change

**Comment by eric-wieser** [comment]
Or as suggested on Zulip, maybe changing `LinearIndependent` to take `ι` explicitly would resolve the problem more cleanly.

**Comment by apnelson1** [comment]
> ```lean
> /-- An alias for `Subtype.val` from sets, which takes `s` as an explicit argument to make it clear what the set in question is. -/
> abbrev Set.incl (s : Set α) : s → α := Subtype.val
> ```

Thanks! I missed this comment, and this looks nicer to me than the alternatives discussed on Zulip, if you're ok with it.

File: Mathlib/Data/Set/Function.lean [hunk 1]
```diff
@@ -76,6 +76,8 @@ theorem image_restrict (f : α → β) (s t : Set α) :
   rw [restrict_eq, image_comp, image_preimage_eq_inter_range, Subtype.range_coe]
 #align set.image_restrict Set.image_restrict
 
+@[simp] theorem restrict_id (s : Set α) : s.restrict id = Subtype.val := rfl
```
Comment (eric-wieser) [inline Mathlib/Data/Set/Function.lean:79]:
```suggestion
@[simp] theorem restrict_id (s : Set α) : s.restrict id = incl := rfl
```

File: Mathlib/LinearAlgebra/AffineSpace/Independent.lean [hunk 2]
```diff
@@ -481,6 +481,10 @@ theorem exists_nontrivial_relation_sum_zero_of_not_affine_ind {t : Finset V}
       by_cases hx : x ∈ t <;> simp [hx]
     all_goals
       simp only [Finset.sum_dite_of_true fun _ h => h, Finset.mk_coe, hwt, hw]
+    · simpa using hwt
+    simpa using hw
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/AffineSpace/Independent.lean:485]: Where did these come from?

File: Mathlib/Data/Set/Basic.lean [hunk 3]
```diff
@@ -2859,6 +2861,10 @@ theorem inclusion_le_inclusion [Preorder α] {s t : Set α} (h : s ⊆ t) {x y :
 theorem inclusion_lt_inclusion [Preorder α] {s t : Set α} (h : s ⊆ t) {x y : s} :
     inclusion h x < inclusion h y ↔ x < y := Iff.rfl
 
+/-- An alias for `Subtype.val` from sets, which takes `s` as an explicit argument to make it clear
+  what the set in question is. -/
+abbrev incl (s : Set α) : s → α := Subtype.val
+
```
Comment (eric-wieser) [inline Mathlib/Data/Set/Basic.lean:2867]:
```suggestion

@[simp] theorem incl_apply (s : Set α) (x : s) : Set.incl s x = x := rfl 
```

File: Mathlib/Data/Set/Basic.lean [hunk 4]
```diff
@@ -2859,6 +2861,10 @@ theorem inclusion_le_inclusion [Preorder α] {s t : Set α} (h : s ⊆ t) {x y :
 theorem inclusion_lt_inclusion [Preorder α] {s t : Set α} (h : s ⊆ t) {x y : s} :
     inclusion h x < inclusion h y ↔ x < y := Iff.rfl
 
+/-- An alias for `Subtype.val` from sets, which takes `s` as an explicit argument to make it clear
+  what the set in question is. -/
+abbrev incl (s : Set α) : s → α := Subtype.val
```
Comment (eric-wieser) [inline Mathlib/Data/Set/Basic.lean:2866]:
```suggestion
/-- An abbreviation for `Subtype.val` from sets, which takes `s` as an explicit argument to make
it clear what the set in question is.

See also `Set.inclusion`, for when the codomain is also a set.
-/
abbrev incl (s : Set α) : s → α := Subtype.val
```
